rules/CodingStyle/Rector/FuncCall/FirstClassCallableRector.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\FuncCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Arg;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Name;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\VariadicPlaceholder;
+use PHPStan\Analyser\Scope;
+use Rector\Rector\AbstractScopeAwareRector;
+use Rector\ValueObject\PhpVersion;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use ReflectionException;
+use ReflectionFunction;
+use ReflectionNamedType;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\CodingStyle\Rector\FuncCall\FirstClassCallableRector\FirstClassCallableRectorTest
+ */
+final class FirstClassCallableRector extends AbstractScopeAwareRector implements MinPhpVersionInterface
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        // see RFC https://wiki.php.net/rfc/first_class_callable_syntax
+        return new RuleDefinition('Upgrade string callback functions to first class callable', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    public function run(array $data)
+    {
+        return array_map('trim', $data);
+    }
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    public function run(array $data)
+    {
+        return array_map(trim(...), $data);
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [FuncCall::class];
+    }
+
+    public function refactorWithScope(Node $node, Scope $scope): ?FuncCall
+    {
+        if (! $node instanceof FuncCall) {
+            return null;
+        }
+
+        if (! $node->name instanceof Name) {
+            return null;
+        }
+
+        if ($node->isFirstClassCallable()) {
+            return null;
+        }
+
+        $functionName = (string) $this->getName($node);
+
+        try {
+            $reflectionFunction = new ReflectionFunction($functionName);
```

Comment:
I am aware, you just moved this code arround: this code should not use native php-src reflection but the PHPStan ReflectionProvider (like we do in other classes)

Reply:
@staabm can't work without it, see original PR for it

Reply:
https://github.com/rectorphp/rector-src/pull/6027#pullrequestreview-2137659835

Reply:
I see. I will look into it after this one is merged